packages/flutter_migrate/lib/src/compute.dart

```diff
@@ -0,0 +1,1058 @@
+  'test', // Typically user owned and flutter-side changes are not relevant.
+];
+
+final Iterable<String> canonicalizedSkippedFiles = _skippedFiles.map<String>(
+  (String path) => canonicalize(path),
+);
+// final String canonicalizedLocalPath = canonicalize(localPath);
+// if (canonicalizedSkippedFiles.contains(canonicalizedLocalPath)) {
+//   return true;
+// }
+
+// Returns true for paths relative to the project root that should be skipped
+// completely by the migrate tool.
+bool _skipped(String localPath, FileSystem fileSystem,
+    {Set<String>? skippedPrefixes}) {
+  final String canonicalizedLocalPath = canonicalize(localPath);
+  final Iterable<String> canonicalizedSkippedFiles =
+      _skippedFiles.map<String>((String path) => canonicalize(path));
+  if (canonicalizedSkippedFiles.contains(canonicalizedLocalPath)) {
+    return true;
+  }
+  final Iterable<String> canonicalizedSkippedDirectories =
+      _skippedDirectories.map<String>((String path) => canonicalize(path));
+  for (final String dir in canonicalizedSkippedDirectories) {
+    if (canonicalizedLocalPath.startsWith('$dir${fileSystem.path.separator}')) {
+      return true;
+    }
+  }
+  if (skippedPrefixes != null) {
+    final Iterable<String> canonicalizedSkippedPrefixes =
+        _skippedFiles.map<String>((String path) => canonicalize(path));
+    return canonicalizedSkippedPrefixes.any((String prefix) =>
+        canonicalizedLocalPath
+            .startsWith('${canonicalize(prefix)}${fileSystem.path.separator}'));
+  }
+  return false;
+}
+
+// File extensions that the tool will not attempt to merge. Changes
+// in files with these extensions will be accepted wholesale.
+//
+// The executables and binaries in this list are not meant to be
+// comprehensive and need only cover the files that are generated
+// in `flutter create` as only files generated by the template
+// will be attempted to be merged.
+const List<String> _doNotMergeFileExtensions = <String>[
+  // Don't merge image files
+  '.bmp',
+  '.gif',
+  '.jpg',
+  '.jpeg',
+  '.png',
+  '.svg',
+  // Don't merge compiled artifacts and executables
+  '.dll',
+  '.exe',
+  '.jar',
+  '.so',
+];
+
+// These files should always go through the migrate process as
+// they are either integral to the migrate process or we expect
+// new versions of this file to always be desired.
+const Set<String> _alwaysMigrateFiles = <String>{
+  '.metadata', // .metadata tracks key migration information.
+  'android/gradle/wrapper/gradle-wrapper.jar',
+  // Always add .gitignore back in even if user-deleted as it makes it
+  // difficult to migrate in the future and the migrate tool enforces git
+  // usage.
+  '.gitignore',
+};
+
+/// False for files that should not be merged. Typically, images and binary files.
+bool _mergable(String localPath) {
+  for (final String ext in _doNotMergeFileExtensions) {
+    if (localPath.endsWith(ext) && !_alwaysMigrateFiles.contains(localPath)) {
```

Comment:
There's no need to re-check `_alwaysMigrateFiles` in the loop, since that expression doesn't use `ext`.

It looks like this whole function could simplify to:
```
return _alwaysMigrateFiles.contains(localPath) || ! _doNotMergeFileExtensions.any((String ext) => localPath.endsWith(ext));
```